CollapseLauncher/Classes/InstallManagement/BaseClass/InstallManagerBase.cs

```diff
@@ -481,17 +483,168 @@ public async Task MoveGameLocation()
+                                foldersToKeepInData = null;
+                                break;
+
+                            case GameType.StarRail:
+                                gameDataFolderName = "StarRail_Data";
+                                foldersToDelete = new string[] { "AntiCheatExpert" };
+                                filesToDelete = new string[] { "ACE-BASE.sys", "GameAssembly.dll", "pkg_version", "config.ini", "^StarRail.*", "^Unity.*" };
+                                foldersToKeepInData = new string[] { "ScreenShots" };
+                                break;
+
+                            case GameType.Genshin:
+                                switch (ConfigV2Store.CurrentConfigV2GameRegion)
+                                {
+                                    case "Global":
+                                        gameDataFolderName = "GenshinImpact_Data";
+                                        foldersToDelete = new string[] { "GenshinImpact_Data" };
+                                        filesToDelete = new string[] { "HoYoKProtect.sys", "pkg_version", "GenshinImpact.exe", "UnityPlayer.dll", "config.ini", "^mhyp.*", "^Audio.*" };
+                                        foldersToKeepInData = null;
+                                        break;
+
+                                    case "Mainland China":
+                                        gameDataFolderName = "YuanShen_Data";
+                                        foldersToDelete = new string[] { "YuanShen_Data" };
+                                        filesToDelete = new string[] { "HoYoKProtect.sys", "pkg_version", "YuanShen.exe", "UnityPlayer.dll", "config.ini", "^mhyp.*", "^Audio.*" };
+                                        foldersToKeepInData = null;
+                                        break;
+
+                                    default:
+                                        LogWriteLine($"Unknown GI Game Region! : {ConfigV2Store.CurrentConfigV2GameRegion}", LogType.Error, true);
+                                        break;
+                                }
+                                break;
+
+                            case GameType.Zenless:
+                                LogWriteLine($"Cannot uninstall game: Zenless uninstall method is not yet implemented!", LogType.Error, true);
+                                return false;
+
+                            default:
+                                LogWriteLine($"Cannot uninstall game: GameType is Unknown!", LogType.Error, true);
+                                return false;
+                        }
+
+                        // Cleanup Game_Data folder while keeping whatever specified in foldersToKeepInData
+                        foreach (string folderGameData in Directory.GetDirectories(Path.Combine(GameFolder, gameDataFolderName)))
```

Comment:
The ``Directory.GetDirectories()`` can be replaced with ``Directory.EnumerateDirectories()`` since we only need to enumerate the existing list of the directory without allocating the ``string[]`` to memory from ``Directory.GetDirectories()``

This can be applied to replace ``Directory.GetFiles()`` with ``Directory.EnumerateFiles()`` codes below too